transports/quic/src/endpoint.rs

```diff
@@ -402,18 +402,41 @@ where
 
         if me.event_slot.is_none() && me.incoming_slot.is_none() {
             let mut metas = [RecvMeta::default(); BATCH_SIZE];
-            let mut iovs = MaybeUninit::<[IoSliceMut; BATCH_SIZE]>::uninit();
-            me.recv_buf
-                .chunks_mut(me.recv_buf.len() / BATCH_SIZE)
-                .enumerate()
-                .for_each(|(i, buf)| unsafe {
-                    iovs.as_mut_ptr()
-                        .cast::<IoSliceMut>()
-                        .add(i)
-                        .write(IoSliceMut::new(buf));
-                });
-            let mut iovs = unsafe { iovs.assume_init() };
-            while let Poll::Ready(result) = me.socket.poll_recv(cx, &mut iovs, &mut metas) {
+            let mut iovs = unsafe {
+                // TODO: use MaybeUninit::uninit_array when stabilized
+                // SAFETY: An uninitialized `[MaybeUninit<_>; LEN]` is valid.
+                MaybeUninit::<[MaybeUninit<IoSliceMut>; BATCH_SIZE]>::uninit().assume_init()
+            };
+            // SAFETY: safe as long as iovs do not outlive recv_buf
+            // and there exists only one iovs initialized with the given recv_buf.
+            unsafe fn init_iovs<'a, 'b>(
+                iovs: &'b mut [MaybeUninit<IoSliceMut>],
+                recv_buf: &'a mut [u8],
+            ) -> &'b mut [IoSliceMut<'b>] {
+                // every iovs elem must be initialized with an according elem from buf chunks
+                debug_assert_eq!(
```

Comment:
```suggestion
                assert_eq!(
```

I would rather leave this assert on in release mode as a precaution.

Reply:
Yes, this is cheap since ChunksMut implements ExactSizeIterator.